akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -138,9 +138,14 @@ private[akka] final case class WidenedInterceptor[O, I](matcher: PartialFunction
   import BehaviorInterceptor._
 
   override def isSame(other: BehaviorInterceptor[Any, Any]): Boolean = other match {
-    // can only be elimintated if it is the same partial function
+    // If they use the same pf instance we can allow it, to have one way to workaround defining
+    // "recursive" narrowed behaviors.
     case WidenedInterceptor(`matcher`) ⇒ true
-    case _                             ⇒ false
+    case WidenedInterceptor(otherMatcher) ⇒
+      // there is no safe way to allow this
+      throw new IllegalStateException("Widen can only be used one time in the same behavior stack. " +
```

Comment:
will this result in infinite restart loop if unlimited `SupervisorStrategy.restart` is used?

should we have protection against restart loops by treating unlimited -1 as not really unlimited but with a very hight limit such as 10000 restarts within 1 second?

wdyt @chbatey

Reply:
Good, then it would have to be a new message triggering the exception again. Probably ok to leave the unlimited. It’s not a problem we have heard of for untyped.